x-pack/plugins/lens/public/editor_frame_plugin/editor_frame/editor_frame.tsx

```diff
@@ -92,9 +118,40 @@ export function EditorFrame(props: EditorFrameProps) {
     [datasourcePublicAPI, state.visualization.activeId, state.visualization.state]
   );
 
-  if (state.datasource.activeId && !state.datasource.isLoading) {
+  const datasource =
+    state.datasource.activeId && !state.datasource.isLoading
+      ? props.datasourceMap[state.datasource.activeId]
+      : undefined;
+
+  const visualization = state.visualization.activeId
+    ? props.visualizationMap[state.visualization.activeId]
+    : undefined;
+
+  if (datasource) {
     return (
       <FrameLayout
+        navPanel={
+          <nav>
+            <EuiButton
+              fill
+              onClick={() =>
+                save({
+                  datasource,
+                  dispatch,
+                  visualization,
+                  state,
+                  redirectTo: props.redirectTo,
+                  store: props.store,
+                })
+              }
+              disabled={state.saving}
```

Comment:
We could disable this button as long as there isn't an active datasource _and_ visualization. Then you don't need the branches in `save.ts` for these cases.

Reply:
That's probably a reasonable idea. I can't think of a good reason to save a partially assembled Lens visualization.